### PR TITLE
Improve nesting

### PR DESCRIPTION
**Pull request by Szegoo**
Closes: #223

### CODE REVIEW

**Comment by Szegoo** [comment]
@ilionic PTAL.

**Comment by Szegoo** [comment]
Maybe a bit early for a review, will ping you when it will be ready :)

**Comment by Szegoo** [comment]
@ilionic This is ready for review, other than the `burn_nft` I couldn't find any other functions that have recursion. Are these the only ones?

**Comment by ilionic** [comment]
Thanks @Szegoo. That's correct, recursive calls are only related to burning and root owner lookup.
Also please check unit tests: 
```
error[E0046]: not all trait items implemented, missing: `NestingBudget`
   --> pallets/rmrk-market/src/mock.rs:107:1
    |
107 | impl pallet_rmrk_core::Config for Test {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `NestingBudget` in implementation
```

File: traits/src/nft.rs [hunk 1]
```diff
@@ -73,8 +73,6 @@ pub struct NftChild {
 /// Abstraction over a Nft system.
 #[allow(clippy::upper_case_acronyms)]
 pub trait Nft<AccountId, BoundedString, BoundedResourceVec> {
-	type MaxRecursions: Get<u32>;
```
Comment (Szegoo) [inline traits/src/nft.rs:76]: I don't think we need `MaxRecursions`.

**Comment by HashWarlock** [comment]
Do we prevent this scenario when sending/minting NFTs to NFTs? If we are able to get a `Budget` from `lookup_root_owner`, could we prevent this scenario from happening?

Nested-Logic-Q

**Comment by Szegoo** [comment]
> If we are able to get a `Budget` from `lookup_root_owner`, could we prevent this scenario from happening?

Not sure I understand this totally correctly but did you mean that we should somehow be able to get the `Budget` that is needed to look up the owner?

**Review by HashWarlock — APPROVED** [review]
LGTM

**Pull request merged**